src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -136,22 +136,22 @@ CompilationStartAnalysisContext context
 				SyntaxKind.LocalFunctionStatement
 			);
 
-			context.RegisterSyntaxNodeAction(
+			context.RegisterSymbolAction(
 				ctx => AnalyzeConflictingImmutabilityOnTypeParameters(
 					ctx,
+					(INamedTypeSymbol)ctx.Symbol,
 					annotationsContext
 				),
-				SyntaxKind.TypeParameter
+				SymbolKind.NamedType
 			);
 
-			context.RegisterSyntaxNodeAction(
+			context.RegisterSymbolAction(
 				ctx => AnalyzeConflictingImmutabilityOnMember(
 					ctx,
+					(INamedTypeSymbol)ctx.Symbol,
 					annotationsContext
 				),
-				SyntaxKind.ClassDeclaration,
-				SyntaxKind.InterfaceDeclaration,
-				SyntaxKind.StructDeclaration
+				SymbolKind.NamedType
 			);

```

Comment:
There's a few NamedType symbol registrations, but leaving refactoring or combining for a later exercise